v1/client_test.go

```diff
@@ -50,6 +60,386 @@ func TestWriteSchemaCall(t *testing.T) {
+	_, beatrice, postOne, _, err := WriteTestTuples(client)
+	require.NoError(err)
+
+	// Likes Harry Potter
+	likesContext, err := structpb.NewStruct(map[string]any{"likes": true})
+	require.NoError(err)
+	firstResponse, err := client.PermissionsServiceClient.CheckPermission(context.Background(), &v1.CheckPermissionRequest{
+		Resource:    postOne,
+		Permission:  "view_as_fan",
+		Subject:     beatrice,
+		Consistency: fullyConsistent,
+		Context:     likesContext,
+	})
+	require.NoError(err)
+	require.Equal(v1.CheckPermissionResponse_PERMISSIONSHIP_HAS_PERMISSION, firstResponse.Permissionship)
+
+	// No longer likes Harry Potter
+	dislikesContext, err := structpb.NewStruct(map[string]any{"likes": false})
+	require.NoError(err)
+	secondResponse, err := client.PermissionsServiceClient.CheckPermission(context.Background(), &v1.CheckPermissionRequest{
+		Resource:    postOne,
+		Permission:  "view_as_fan",
+		Subject:     beatrice,
+		Consistency: fullyConsistent,
+		Context:     dislikesContext,
+	})
+	require.NoError(err)
+	require.Equal(v1.CheckPermissionResponse_PERMISSIONSHIP_NO_PERMISSION, secondResponse.Permissionship)
+
+	// Fandom is in question
+	require.NoError(err)
+	thirdResponse, err := client.PermissionsServiceClient.CheckPermission(context.Background(), &v1.CheckPermissionRequest{
+		Resource:    postOne,
+		Permission:  "view_as_fan",
+		Subject:     beatrice,
+		Consistency: fullyConsistent,
+	})
+	require.NoError(err)
+	require.Equal(v1.CheckPermissionResponse_PERMISSIONSHIP_CONDITIONAL_PERMISSION, thirdResponse.Permissionship)
+	require.Contains(thirdResponse.PartialCaveatInfo.MissingRequiredContext, "likes")
+}
+
+func TestLookupResources(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	client := SutClient(t)
+	defer client.Conn.Close()
+
+	err := WriteTestSchema(client)
+	require.NoError(err)
+	emilia, _, postOne, postTwo, err := WriteTestTuples(client)
+	require.NoError(err)
+
+	// NOTE: setting a page limit and then cursoring over that limit is entirely overkill
+	// for this case, where we know how many results we're expecting. This is meant as an
+	// example to demonstrate a real-world lookupResources usage.
+	pageLimit := 50
+	// The page buffer is where we'll store individual results from the stream
+	pageBuffer := make([]string, 0, pageLimit)
+	// Where the result buffer is where we'll concatenate the page buffers together
+	// to get a final set of results
+	resultBuffer := make([]string, 0)
+
+	for {
+		response, err := client.PermissionsServiceClient.LookupResources(context.Background(), &v1.LookupResourcesRequest{
+			ResourceObjectType: "post",
+			Permission:         "write",
+			Subject:            emilia,
+			Consistency:        fullyConsistent,
+		})
+		require.NoError(err)
+
+		resultCount := 0
+	stream:
+		for {
+			item, err := response.Recv()
+			switch {
+			case errors.Is(err, io.EOF):
+				break stream
+			case err != nil:
+				require.NoError(err)
+			default:
+				resultCount += 1
```

Comment:
I don't think you need `resultCount` since you can just get `len(pageBuffer)`

Reply:
Ah that's a good point. I also found what would have been a bug if someone had copy-pasted the code because I would have been reusing the page buffer.